lib/services/fb_login.dart

```diff
@@ -2,12 +2,26 @@ import "dart:async";
 import "dart:convert" show json;
 import 'package:http/http.dart' as http;
 
+import '../models/user_model.dart';
+import '../config.dart';
 
+const url = '$apiAuthorityUrl/users';
 
-Future loadUserProfile(dynamic accessToken) async {
-  String queryFields = "name,first_name,last_name,email,location,profile_pic";
+Future<User> loadUserProfile(dynamic accessToken) async {
+  String queryFields = "name,first_name,last_name,email,location,picture";
 
   final graphResponse = await http.get(
       "https://graph.facebook.com/v4.0/me?fields=$queryFields&access_token=$accessToken");
-  return json.decode(graphResponse.body);
+  final jsonProfile = json.decode(graphResponse.body);
+  return User.fromJson(jsonProfile);
+}
+
+Future saveUserProfile(User user) async {
```

Comment:
Why don't you specify the return type of the future? I was expecting something like Future<int>?

Reply:
I will add a specific type for Future